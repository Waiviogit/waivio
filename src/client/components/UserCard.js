import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import FollowButton from '../widgets/FollowButton';
import Avatar from '../components/Avatar';

import './UserCard.less';

const UserCard = ({ user, alt, showFollow, unfollow, follow }) => {
  const cardView = showFollow ? 'UserCard__left' : 'UserCard__sidebar';
  const weightBlock = showFollow ? 'UserCard__alt BlockWeight' : 'UserCard__short';
  const followersCountBlock = showFollow ? 'reblogged' : 'rebloggedFollowersNone';

  return (
    user && (
      <div className="UserCard">
        <div className={cardView}>
          <div className="UserCard__wrap">
            <Link to={`/@${user.name}`}>
              <Avatar username={user.name} size={40} />
            </Link>
            <Link to={`/@${user.name}`}>
              <span className="username">{user.name}</span>
            </Link>
            {user.admin && (
              <span>
                &nbsp;(<span className="UserCard__status">admin</span>)
              </span>
            )}
          </div>
          {alt && <span className={weightBlock}>{alt}</span>}
          <span className={followersCountBlock}>
            {Boolean(user.followers_count) && `Â· ${user.followers_count} `}
          </span>
        </div>
        {showFollow && (
          <FollowButton
            unfollowUser={unfollow}
            followUser={follow}
            following={user.youFollows}
            user={user}
            followingType="user"
            secondary
          />
        )}
      </div>
    )
  );
};

UserCard.propTypes = {
  user: PropTypes.shape(),
  alt: PropTypes.node,
  showFollow: PropTypes.bool,
  unfollow: PropTypes.func,
  follow: PropTypes.func,
};

UserCard.defaultProps = {
  alt: '',
  user: {},
  showFollow: true,
  authUser: '',
  unfollow: () => {},
  follow: () => {},
};

export default UserCard;
