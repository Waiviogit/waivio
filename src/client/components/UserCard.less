@import (reference) '../styles/custom.less';

.UserCard {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-bottom: 14px;
  padding-top: 14px;
  border-bottom: @base-border;

  .Follow {
    margin-left: 8px;
  }

  &:first-child {
    padding-top: 0;
  }

  & > &__left {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;

    & > a:nth-child(2) {
      margin-left: 8px;
    }

    .username {
      margin-left: 8px;
    }

    .ant-tag {
      margin-right: 0;
    }
  }

  &__wrap {
    display: flex;
    align-items: center;
  }
}

div[data-dir='rtl'] .UserCard {
  & > &__left {
    & > a:nth-child(2) {
      margin-left: 0;
      margin-right: 8px;
    }
  }

  &__alt {
    margin-left: 0;
    margin-right: 8px;
  }
}

@import 'UserCard-nightmode';
