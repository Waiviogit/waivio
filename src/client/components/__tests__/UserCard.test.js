import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { IntlProvider } from 'react-intl';

import UserCard from '../UserCard';

describe('<UserCard />', () => {
  const mockStore = configureStore();
  let props;
  let wrapper;
  let store;

  beforeEach(() => {
    props = {
      user: {},
      unfollow: () => {},
      follow: () => {},
      showFollow: true,
      alt: <span />,
    };
    store = mockStore();

    wrapper = shallow(
      <Provider store={store}>
        <IntlProvider locale="en">
          <UserCard {...props} />
        </IntlProvider>
        ,
      </Provider>,
    );
  });

  it('should render without exploding', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
