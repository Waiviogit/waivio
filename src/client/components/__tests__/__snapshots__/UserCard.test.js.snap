// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> should render without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <IntlProvider
      locale="en"
    >
      <UserCard
        alt={<span />}
        authUser=""
        follow={[Function]}
        showFollow={true}
        unfollow={[Function]}
        user={Object {}}
      />
    </IntlProvider>
    ,
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <IntlProvider
          locale="en"
        >
          <UserCard
            alt={<span />}
            authUser=""
            follow={[Function]}
            showFollow={true}
            unfollow={[Function]}
            user={Object {}}
          />
        </IntlProvider>,
        ",",
      ],
      "value": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <UserCard
            alt={<span />}
            authUser=""
            follow={[Function]}
            showFollow={true}
            unfollow={[Function]}
            user={Object {}}
          />,
          "locale": "en",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": <span />,
            "authUser": "",
            "follow": [Function],
            "showFollow": true,
            "unfollow": [Function],
            "user": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      ",",
    ],
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
        "displayName": "ReactRedux",
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <IntlProvider
            locale="en"
          >
            <UserCard
              alt={<span />}
              authUser=""
              follow={[Function]}
              showFollow={true}
              unfollow={[Function]}
              user={Object {}}
            />
          </IntlProvider>,
          ",",
        ],
        "value": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "subscription": Subscription {
            "handleChangeWrapper": [Function],
            "listeners": Object {
              "notify": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": null,
          },
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <UserCard
              alt={<span />}
              authUser=""
              follow={[Function]}
              showFollow={true}
              unfollow={[Function]}
              user={Object {}}
            />,
            "locale": "en",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": <span />,
              "authUser": "",
              "follow": [Function],
              "showFollow": true,
              "unfollow": [Function],
              "user": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        ",",
      ],
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
          "displayName": "ReactRedux",
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
